src/Microsoft.NetCore.Analyzers/Core/Security/DoNotCallDangerousMethodsInDeserialization.cs

```diff
@@ -195,11 +233,12 @@ public sealed override void Initialize(AnalysisContext context)
                     // <param name="methodSymbol">The symbol of the method to be analyzed</param>
                     // <param name="visited">All the method has been analyzed</param>
                     // <param name="results">The result is organized by &lt;method to be analyzed, dangerous method it calls&gt;</param>
-                    void FindCalledDangerousMethod(IMethodSymbol methodSymbol, HashSet<IMethodSymbol> visited, Dictionary<IMethodSymbol, HashSet<IMethodSymbol>> results)
+                    void FindCalledDangerousMethod(ISymbol methodSymbol, HashSet<ISymbol> visited, Dictionary<ISymbol, HashSet<ISymbol>> results)
                     {
                         if (visited.Add(methodSymbol))
                         {
-                            results.Add(methodSymbol, new HashSet<IMethodSymbol>());
+                            results.Add(methodSymbol, new HashSet<ISymbol>());
+                            Debug.Assert(callGraph.Keys.Contains(methodSymbol), methodSymbol.Name + "was not present in the dictionary.");
 
                             foreach (var child in callGraph[methodSymbol].Keys)
```

Comment:
Instead of just a Debug.Assert(), we could also gracefully handle missing keys in retail builds.

```csharp
if (!callGraph.TryGetValue(methodSymbol, out var calledMethods))
{
    Debug.Fail(methodSymbol.Name + " was not found in callGraph);
    return;
}

foreach (var child in calledMethods.Keys)
```

Reply:
+1